src/plugins/query_enhancements/public/query_assist/utils/get_is_summary_agent.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { getAssistantClient } from '../../services/index';
+import { DATA2SUMMARY_AGENT_CONFIG_ID } from './constant';
+
+export async function getIsSummaryAgent(dataSourceID: string) {
+  const assistantClient = getAssistantClient();
```

Comment:
imo if it's easier to add this code directly into the query enhancements and do the inverse of this where you access the query enhancements client. that would be safer option as the bundle will always be distributed for query enhancements

Reply:
but there might be an even better solution that is scalable like updating a service that lets you register a summary provider and then this plugin just needs to access it from the data plugin

Reply:
but the desired solution is that the reference to this banner is free real estate for any plugin. you should be able to do this all from within the query assist plugin directly